source/CAS/Client.php

```diff
@@ -1082,6 +1103,144 @@ public function getChangeSessionID()
         return $this->_change_session_id;
     }
 
+    /**
+     * Set the session handler.
+     *
+     * @param \SessionHandlerInterface $sessionHandler
+     *
+     * @return bool
+     */
+    public function setSessionHandler(\SessionHandlerInterface $sessionHandler)
+    {
+        $this->_sessionHandler = $sessionHandler;
+        return session_set_save_handler($this->_sessionHandler, true);
+    }
+
+    /**
+     * Get a session value using the given key.
+     *
+     * @param string $key
+     * @param mixed  $default default value if the key is not set
+     *
+     * @return mixed
+     */
+    protected function getSessionValue($key, $default = null)
+    {
+        $this->validateSession($key);
+
+        if (isset($_SESSION[static::PHPCAS_SESSION_PREFIX][$key])) {
+            return $_SESSION[static::PHPCAS_SESSION_PREFIX][$key];
+        }
+
+        return $default;
+    }
+
+    /**
+     * Determine whether a session value is set or not.
+     *
+     * @param string $key
+     *
+     * @return bool
+     */
+    protected function hasSessionValue($key)
+    {
+        $this->validateSession($key);
+
+        return (isset($_SESSION[static::PHPCAS_SESSION_PREFIX][$key])
+        && !empty($_SESSION[static::PHPCAS_SESSION_PREFIX][$key]));
```

Comment:
```suggestion
        return (isset($_SESSION[static::PHPCAS_SESSION_PREFIX][$key]);
```

If an empty check is needed, `empty(getSessionValue($key))` should be used instead.